### DISCUSSION TITLE
Content Delivery API: Suggestion about to retrieve protected data

### DISCUSSION THREAD

**Discussion by biapar** [umbraco/Umbraco-CMS#14483] [Features and ideas]
Hi,
Today, the lack of the support that Delivery API does not support authentication for members is a problem and so we cannot use Umbraco as a very headless CMS like the others.
Until now, I'm using U10 LTS with an headless custom package Nikcio.UHeadless. 
To retrieve the protected content of a member, I create before a group generic and a group for the member.
So I protect the node with generic + member or only member and the parent with the generic group.
In the request , I pass the member ID + email and check if the permission on the node is ok or not.

**Comment by lassefredslund** [comment]
HI @biapar, 
Just FYI: We have started implementing Member Authentication for the Content Delivery API, and expect to release it in a minor of V12.

**Comment by biapar** [comment]
Thank you. Interesting. Where can I read the code in development?

**Reply by bergmania** [reply]
https://github.com/umbraco/Umbraco-CMS/pull/14730

It's currently targeting V14, as we need to ensure it plays perfectly with the upcoming management API. The idea is to backport it to V12 when it's done
